prometheus_client/values.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import unicode_literals
+
+import os
+from threading import Lock
+
+
+class MutexValue(object):
+    '''A float protected by a mutex.'''
+
+    _multiprocess = False
+
+    def __init__(self, typ, metric_name, name, labelnames, labelvalues, **kwargs):
+        self._value = 0.0
+        self._lock = Lock()
+
+    def inc(self, amount):
+        with self._lock:
+            self._value += amount
+
+    def set(self, value):
+        with self._lock:
+            self._value = value
+
+    def get(self):
+        with self._lock:
+            return self._value
+
+
+def MultiProcessValue(_pidFunc=os.getpid):
+    from .mmap_dict import MmapedDict, mmap_key
```

Comment:
Imports should be at the top level, this way has been associated with deadlocks

Reply:
The idea here was that the whole mmap functionality wouldn't even be imported unless a MultiProcessValue is required, but I don't mind changing it.

Curious though -- what sort of deadlocks do you mean? In what situations would those maybe happen? 😲